apm-agent-plugins/apm-api-plugin/src/main/java/co/elastic/apm/agent/pluginapi/ElasticApmApiInstrumentation.java

```diff
@@ -162,4 +163,25 @@ public static void setServiceInfoForClassLoader(@Advice.Argument(0) @Nullable Cl
             }
         }
     }
+
+    public static class ConfigInstrumentation extends ElasticApmApiInstrumentation {
+        public ConfigInstrumentation() {
+            super(named("getConfig"));
+        }
+
+        public static class AdviceClass {
+            @Nullable
+            @Advice.AssignReturned.ToReturned
+            @Advice.OnMethodExit(suppress = Throwable.class, inline = false)
+            public static Object getConfig(@Advice.Argument(0) @Nullable String key) {
+                try {
+                    Object value= GlobalTracer.getTracerImpl().getConfigurationRegistry().getConfigurationOptionByKey(key).getValue();
+                    // should we return a special value for null - which means non-existent entry?
+                    return value;
+                } catch (NullPointerException e) {
+                    return null;
+                }
```

Comment:
While we usually try to be very kind to our end-users, I would say that here we should throw an `IllegalArgumentException` if there is no matching config option. If we add the expected returned type as an argument, then we can also check the returned type and also throw an error if it's invalid.
Both cases would be bugs on the calling code, so we'd better throw exceptions early to prevent bugs from compounding later.